improc/photometry.py

```diff
@@ -253,18 +262,16 @@ def iterative_photometry(
             areas[j] = np.nansum(mask)  # save the number of pixels in the aperture
 
             # get an offset annulus to get a local background estimate
-            inner = get_circle(radius=annulus[0], imsize=nandata.shape[0]).get_image(reposition_cx, reposition_cy)
-            outer = get_circle(radius=annulus[1], imsize=nandata.shape[0]).get_image(reposition_cx, reposition_cy)
+            inner = get_circle(radius=annulus[0], imsize=nandata.shape[0], soft=False).get_image(reposition_cx, reposition_cy)
+            outer = get_circle(radius=annulus[1], imsize=nandata.shape[0], soft=False).get_image(reposition_cx, reposition_cy)
             annulus_map = outer - inner
 
             # background and variance only need to be calculated once (they are the same for all apertures)
             # but moments/centroids can be calculated for each aperture, but we will only want to save one
             # so how about we use the smallest one?
             if j == 0:  # smallest aperture only
-                # TODO: consider replacing this with a hard-edge annulus and do median or sigma clipping on the pixels
-                background = np.nansum(nandata * annulus_map) / np.nansum(annulus_map)  # b/g per pixel
+                background = np.nansum(sigma_clip(nandata * annulus_map,sigma=5)) / np.nansum(annulus_map)  # sigma-clipped b/g per pixel
                 variance = np.nansum((nandata - background) * annulus_map) ** 2 / np.nansum(annulus_map)  # per pixel
```

Comment:
```suggestion
                background, standard_dev = sigma_clipping(ndata * annulus_map, sigma=5)
                variance = standard_dev ** 2
```

Reply:
The sigma clipping already gives the mean value in the list of pixels it is given, so we don't need to divide by sum of annulus map (the number of pixels). 
It also gives you the standard deviation of those pixels, so we don't need to calculate the variance manually anymore.

Reply:
Also, notice that astropy's `sigma_clip` is not the same as my `sigma_clipping` code...